src-tauri/src/main.rs

```diff
@@ -60,6 +60,8 @@ async fn farming(path: String, reward_address: String, plot_size: u64) -> bool {
             // if there is an error, restart another farmer, and start listening again, in a loop
             loop {
                 let result = error_receiver.recv().await;
+                // probably node restarted, so wait 7 seconds before starting farmer
+                thread::sleep(Duration::from_millis(7000));
```

Comment:
There should be no sleep here. Since farming requires node, you should be able to listen to some notification or observe in some other way that the node node has started. Just doing arbitrary timeout will work sometimes and will fail other times

Reply:
opsie, the same logic is present also when starting farmer in the usual workflow. We will need to address that well...

Reply:
Ideally figure out when RPC starts and subscribe to something that indicates that, in the worst case try to connect to the TCP port where you expect RPC to be available in a loop with a small timeout until it becomes ready.

Reply:
how about the new version? I just pushed some new code about this